h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java

```diff
@@ -13,13 +13,16 @@
 import water.fvec.Frame;
 import water.util.Log;
 import hex.ModelMetrics;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
 import java.util.HashMap;
 import java.util.Map;
 
 import static hex.tree.xgboost.XGBoost.makeDataInfo;
 
 public class XGBoostModel extends Model<XGBoostModel, XGBoostModel.XGBoostParameters, XGBoostOutput> {
-
+  private static final NumberFormat localizedNumberFormatter = DecimalFormat.getNumberInstance();
```

Comment:
Please see documentation to DecimalFormat - there is no guarantee of being thread safe

     * Decimal formats are generally not synchronized.
     * It is recommended to create separate format instances for each thread.
     * If multiple threads access a format concurrently, it must be synchronized
     * externally.